File: class 3/2630.py
import sys

def quad_tree(x, y, n ):
    global blue, white, array
    color = array[y][x]
    state = False

    for i in range(x, x+n):
        if state :
            break

        for j in range(y, y+n):
            
            if color != array[j][i] : 
                quad_tree(x, y, n//2)
                quad_tree(x + n//2, y + n//2, n//2)
                quad_tree(x+ n//2 , y, n//2 )
                quad_tree(x, y+n//2, n//2)
                state = True 
                break

    if not state :
        if array[y][x] == 1:
            blue = blue + 1
        else :
            white = white  +1

    
blue = 0
white = 0
array = []
n = int(input())
for _ in range(n) :
    array.append(list(map(int, input().split())))

quad_tree(0,0,n)
print(white)
print(blue)

File: 백준 문제 풀이 모음/kotlin/class 3/2630.kt

var matrix = MutableList(0)  { MutableList<Int>(0) { 0 } }
var white = 0
var black = 0
fun main() = with(System.`in`.bufferedReader()) {
    val n = readLine().toInt()
    matrix = MutableList(n)  { MutableList<Int>(n) { 0 } }
    for ( i in 0 until n) {
        val input = readLine().split(' ').toList()
        for (j in 0 until n) {
            matrix[i][j] = input[j].toInt()
        }
    }

    divide(n, 0, 0)

    println(white)
    println(black)

}

fun divide(n: Int, rol: Int, col: Int) {

    val first = matrix[rol][col]
    var state = true
    for (i in rol until rol + n) {
        for (j in col until col + n) {
            if (matrix[i][j] != first) {
                state = false
            }
        }
    }

    if (!state) {
        val size = n / 2
        divide(size, rol, col)
        divide(size, rol + size, col)
        divide(size, rol, col + size)
        divide(size, rol + size, col + size)
    } else {
        if (first == 0) {
            white++
        } else {
            black++
        }
    }
}

